### PR TITLE
Add GH Actions CI workflow

### LINKED ISSUE #1679: Move to GitHub actions
We're currently using Travis for our CI, but more and more repositories have moved to using GitHub actions. There are some benefits to this, which include speed and improved integration in the GH website.

### PR DESCRIPTION
**Pull request by cmccandless**
Resolves #1679 

Someone with more node knowledge than me please verify that version 6 is still correct. I also noticed that the Travis config was caching yarn. If this is a requirement for the GH Action, I can research how to accomplish this.

Also let me know if this PR should also remove .travis.yml.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions/setup-node@v1
+        with:
+          node-version: '6'
+      - run: yarn
+      - run: bin/check_required_files_present
+      - run: bin/check_optional
```
Comment (iHiD) [inline .github/workflows/ci.yml:19]:
```suggestion
```

This is about to be removed in the `.travis.yml` via #1677 so probably shouldn't be added here.

**Comment by iHiD** [comment]
You can copy the yarn caching config from the new website: https://github.com/exercism/v3-website/blob/master/.github/workflows/tests.yml#L53-L65

**Comment by cmccandless** [comment]
It's incredible how much faster GH Actions is over Travis.

**Comment by iHiD** [comment]
Yeah, ridiculous isn't it!

**Comment by SleeplessByte** [comment]
Node 6?? That's 2016-04-26. Maybe try 12.18, or better yet: `lts`. 

If it breaks, let's keep 6, and add an issue to upgrade.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+
+      - uses: actions/setup-node@v1
+        with:
+          node-version: '6'
```
Comment (ErikSchierboom) [inline .github/workflows/ci.yml:17]: As @SleeplessByte has also suggested, could you try the LTS version:

```suggestion
          node-version: '12'
```

**Review by ErikSchierboom — APPROVED** [review]
> Also let me know if this PR should also remove .travis.yml.

Yes please!

**Comment by cmccandless** [comment]
@iHiD  or @ErikSchierboom will need to remove Travis CI as a required check for this repository.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  pull_request:
+  push: 
+    branches: 
+      - master
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
```
Comment (SaschaMann) [inline .github/workflows/ci.yml:13]:
```suggestion
      - uses: actions/checkout@v2
```

v2 is much faster than v1

**Comment by iHiD** [comment]
Tell me when this is ready to merge and I'll do that and switch the required check too.

**Review by ErikSchierboom — APPROVED** [review]
@iHiD This is ready to be merged!

**Pull request merged**